app/screens/invite/selection.tsx

```diff
@@ -0,0 +1,512 @@
+                    <View
+                        style={styles.shareLinkButton}
+                        testID='invite.share_link.button'
+                    >
+                        <CompassIcon
+                            name='export-variant'
+                            size={18}
+                            style={styles.shareLinkIcon}
+                        />
+                        <FormattedText
+                            id='invite.shareLink'
+                            defaultMessage='Share link'
+                            style={styles.shareLinkText}
+                        />
+                    </View>
+                </TouchableOpacity>
+            </View>
+            <FormattedText
+                id='invite.sendInvitationsTo'
+                defaultMessage='Send invitations to…'
+                style={styles.searchBarTitleText}
+                onLayout={onLayoutSearchBarTitle}
+                testID='invite.search_bar_title'
+            />
+            <View
+                style={styles.searchBar}
+                onLayout={onLayoutSearchBar}
+            >
+                <FloatingTextInput
+                    autoCorrect={false}
+                    autoCapitalize={'none'}
+                    blurOnSubmit={false}
+                    disableFullscreenUI={true}
+                    enablesReturnKeyAutomatically={true}
+                    placeholder={formatMessage({id: 'invite.searchPlaceholder', defaultMessage: 'Type a name or email address…'})}
+                    onChangeText={handleSearchChange}
+                    keyboardAppearance={getKeyboardAppearanceFromTheme(theme)}
+                    returnKeyType='search'
+                    value={term}
+                    theme={theme}
+                    testID='invite.search_bar_input'
+                />
+            </View>
+            {Object.keys(selectedIds).length > 0 && (
+                <ScrollView
+                    style={styles.selectedItems}
+                    contentContainerStyle={styles.selectedItemsContainer}
+                    testID='invite.selected_items'
+                >
+                    {renderSelectedItems}
+                </ScrollView>
+            )}
+            <Animated.View style={searchListContainerStyle}>
+                <FlatList
+                    data={searchResults}
+                    keyboardShouldPersistTaps='always'
+                    {...keyboardDismissProp}
+                    keyExtractor={keyExtractor}
+                    initialNumToRender={INITIAL_BATCH_TO_RENDER}
+                    ListEmptyComponent={renderNoResults}
+                    maxToRenderPerBatch={INITIAL_BATCH_TO_RENDER + 1}
+                    renderItem={renderItem}
+                    scrollEventThrottle={SCROLL_EVENT_THROTTLE}
+                    style={styles.searchListFlatList}
+                    testID='invite.search_list'
+                />
+            </Animated.View>
+        </View>
+    );
```

Comment:
This component is a little bit too big, can you split it in smaller ones or move some functionality out?

Reply:
Updated